mousestyles/ultradian/__init__.py

```diff
@@ -0,0 +1,381 @@
+    in one dataframe. It combines the results you got from
+    aggregate_movements and aggregate_interval. It will return
+    a dataframe with three variables: mouse, strain, feature and hour.
+
+    Parameters
+    ----------
+    feature :
+        {"AS", "F", "IS", "M_AS", "M_IS", "W"}
+    bin_width : int
+        Number of minutes, the time interval for data aggregation.
+
+    Returns
+    -------
+    pandas.dataframe
+    describe :
+        Column 0: the mouse number (number depends on strain)(0-3)
+        Column 1: the strain of the mouse (0-2)
+        Column 2: hour(numeric values below 24 accourding to bin_width)
+        Column 3: feature values
+
+    Examples
+    --------
+    >>> test = aggregate_data("Distance",20)
+    >>> print(np.mean(test["Distance"]))
+    531.4500177747973
+    """
+    init = pd.DataFrame(columns=["mouse", "strain", "hour", feature])
+    for i in range(3):
+        for j in range(4):
+            if feature == "Distance":
+                tmp = aggregate_movement(strain=i, mouse=j,
+                                         bin_width=bin_width)
+            else:
+                tmp = aggregate_interval(strain=i, mouse=j,
+                                         feature=feature,
+                                         bin_width=bin_width)
+            tmp = pd.DataFrame(list(tmp.values), index=tmp.index)
+            tmp.columns = [feature]
+            tmp["strain"] = i
+            tmp["mouse"] = j
+            tmp["hour"] = tmp.index.hour + tmp.index.minute / 60
+            init = init.append(tmp)
+    return(init)
+
+
+def seasonal_decomposition(strain, mouse, feature, bin_width, period_length):
+    """
+    Apply seasonal decomposition model on the time series
+    of specified strain, mouse, feature and bin_width.
+
+    Parameters
+    ----------
+    strain: int
+        nonnegative integer indicating the strain number
+    mouse: int
+        nonnegative integer indicating the mouse number
+    feature: {"AS", "F", "M_AS", "M_IS", "W", "Distance"}
+        "AS": Active state probalibity
+        "F": Food consumed (g)
+        "M_AS": Movement outside homebase
+        "M_IS": Movement inside homebase
+        "W": Water consumed (g)
+        "Distance": Distance traveled
+    bin_width: int
+        number of minutes, the time interval for data aggregation
+    period_length: int
+        number of hours, usually the significant period
+        length indicated by Lomb-scargle model
+
+    Returns
+    -------
+    res: statsmodel seasonal decomposition object
+        seasonal decomposition result for the mouse.
+        Check the seasonal decomposition plot by res.plot(),
+        seasonl term and trend term by res.seasonal and
+        res.trend separately.
+
+    Examples
+    --------
+    >>> res = seasonal_decomposition(strain=0, mouse=0, feature="W",
+                                     bin_width=30, period_length = 24)
+    """
+    freq = int(period_length * 60 / bin_width)
```

Comment:
Add input check for features (in given set). Add input check for strain mouse (positive integer). Add input check for bin_width(min and max range), etc.